### PR TITLE
Story#590 bottom nav

### PR DESCRIPTION
**Pull request by rejneesh1**

### CODE REVIEW

File: src/pages/bottom-tab/bottom-tab.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<ion-tabs tabsHighlight="true">
+  <ion-tab [root]="homeRoot" tabTitle="Home" tabIcon="home"></ion-tab>
```
Comment (kiranharidas187) [inline src/pages/bottom-tab/bottom-tab.html]: Use translate service for  title

File: src/pages/bottom-tab/bottom-tab.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
+page-bottom-tab {
+  .tabs-md .tabbar {
+    background: #a63936 !important;
+  }
+  .tabs-md .tab-button[aria-selected="true"] .tab-button-icon {
+    color: #fff;
```
Comment (kiranharidas187) [inline src/pages/bottom-tab/bottom-tab.scss]: Use color variables from style.scss

File: src/pages/institution/institution-service.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+          resolve(successData.result);
+        },
+        (error) => {
+          reject();
+        }
+      );
+    });
+  }
+
+  getAssessmentDetails(event, institutionList) {
```
Comment (kiranharidas187) [inline src/pages/institution/institution-service.ts:66]: This is same in programs flow also right?

File: src/pages/institution/institution-solution/institution-solution.ts [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    this.institutionList.entities[this.entityType][
+      this.entityIndex
+    ].solutions.map((sol, solutionIndex) => {
+      this.submissionArr.includes(sol.submissionId)
+        ? (sol.downloaded = true)
+        : null;
+    });
+  }
+
+  goToEcm(solutionIndex, submissionId) {
```
Comment (kiranharidas187) [inline src/pages/institution/institution-solution/institution-solution.ts:89]: Can we reuse this in programs and the institution's flow?

File: src/pages/institution/institution-solution/institution-solution.ts [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+            };
+            this.openAction(assessment, successData, 0);
+            //console.log("else loop");
+          }
+        }
+      })
+      .catch((error) => {});
+  }
+
+  openAction(assessment, aseessmemtData, evidenceIndex) {
```
Comment (kiranharidas187) [inline src/pages/institution/institution-solution/institution-solution.ts:151]: Can we reuse this in programs and the institution's flow?

File: src/pages/institution/institution-solution/institution-solution.ts [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+      _id: assessment._id,
+      name: assessment.name,
+      recentlyUpdatedEntity: assessment.recentlyUpdatedEntity,
+      selectedEvidence: evidenceIndex,
+      entityDetails: aseessmemtData,
+    };
+    this.evdnsServ.openActionSheet(options);
+  }
+
+  getAssessmentDetails(solutionIndex) {
```
Comment (kiranharidas187) [inline src/pages/institution/institution-solution/institution-solution.ts:166]: Can we reuse this in programs and the institution's flow?

File: src/pages/programs/program-solution-entity/program-solution-entity.ts [hunk 7]
```diff
@@ -60,6 +62,68 @@ export class ProgramSolutionEntityPage {
       });
   }
 
+  getSubmissionArr() {
```
Comment (kiranharidas187) [inline src/pages/programs/program-solution-entity/program-solution-entity.ts:65]: Can we reuse this in programs and the institution's flow?

File: src/pages/programs/program-solution-entity/program-solution-entity.ts [hunk 8]
```diff
@@ -60,6 +62,68 @@ export class ProgramSolutionEntityPage {
...
+  //   return this.submissionArr.some(
+  //     (d) => d.solutionId == solutionId && d.entityId == entityId
+  //   );
+  //   // return this.localStorage
+  //   //   .getLocalStorage(storageKeys.submissionIdArray)
+  //   //   .then((allId) => {
+  //   //     return allId.includes(submissionId);
+  //   //   })
+  //   //   .catch((err) => {
+  //   //     this.getAssessmentDetails(entityIndex);
```
Comment (kiranharidas187) [inline src/pages/programs/program-solution-entity/program-solution-entity.ts]: remove commented code if not required

File: src/providers/update-local-school-data/update-local-school-data.ts [hunk 9]
```diff
@@ -223,6 +223,27 @@ export class UpdateLocalSchoolDataProvider {
     }
   }
 
+  async updateSubmissionIdArr(submissionId) {
+    await this.localStorage
+      .getLocalStorage(storageKeys.submissionIdArray)
+      .then(async (arr) => {
+        Array.isArray(submissionId)
```
Comment (kiranharidas187) [inline src/providers/update-local-school-data/update-local-school-data.ts:230]: Why is this check?

Comment (rejneesh1) [reply src/providers/update-local-school-data/update-local-school-data.ts:230]: when migrating then i will pass the array of submissionId other wise string value will come

File: src/pages/bottom-tab/bottom-tab.html (hunk 1 above)
Comment (rejneesh1) [reply src/pages/bottom-tab/bottom-tab.html]: done

File: src/pages/bottom-tab/bottom-tab.scss (hunk 2 above)
Comment (rejneesh1) [reply src/pages/bottom-tab/bottom-tab.scss]: done

File: src/pages/institution/institution-service.ts (hunk 3 above)
Comment (rejneesh1) [reply src/pages/institution/institution-service.ts:66]: here there is difference in getting the ids of program solution and entity 
eg
in program flow entityId is : 
let entityId =
        programs[programIndex].solutions[assessmentIndex].entities[entityIndex]
          ._id;
while in institution flow it is     
 let entityId = institutionList.entities[entityType][entityIndex]._id;

File: src/pages/institution/institution-solution/institution-solution.ts (hunk 4 above)
Comment (rejneesh1) [reply src/pages/institution/institution-solution/institution-solution.ts:89]: same reason as the structure is different i am using two different functions to manipulate the data in each flows

File: src/pages/institution/institution-solution/institution-solution.ts (hunk 5 above)
Comment (rejneesh1) [reply src/pages/institution/institution-solution/institution-solution.ts:151]: same reason as the structure is different i am using two different functions to manipulate the data in each flows

File: src/pages/institution/institution-solution/institution-solution.ts (hunk 6 above)
Comment (rejneesh1) [reply src/pages/institution/institution-solution/institution-solution.ts:166]: same reason as the structure is different i am using two different functions to manipulate the data in each flows

File: src/pages/programs/program-solution-entity/program-solution-entity.ts (hunk 7 above)
Comment (rejneesh1) [reply src/pages/programs/program-solution-entity/program-solution-entity.ts:65]: same reason as the structure is different i am using two different functions to manipulate the data in each flows

**Pull request merged**